File: javascript-agent/src/utils/log.js
export function logInfo(message, fields = {}) {
  console.log(
    JSON.stringify({
      level: "info",
      message,
      ...fields,
      ts: new Date().toISOString(),
    }),
  );
}

export function logError(message, fields = {}) {
  console.error(
    JSON.stringify({
      level: "error",
      message,
      ...fields,
      ts: new Date().toISOString(),
    }),
  );
}

// MCP debug logging — enable with MCP_DEBUG=1 or MCP_DEBUG=true.
// Logs raw tool call details, session lifecycle, and auth envelope metadata.
export function logMcpDebug(message, data) {
  if (process.env.MCP_DEBUG !== "1" && process.env.MCP_DEBUG !== "true") {
    return;
  }
  if (data) {
    console.log("[mcp-debug]", message, JSON.stringify(data));
    return;
  }
  console.log("[mcp-debug]", message);
}

File: python-agent/app/utils/log.py
import json
import os
from datetime import datetime, timezone
from typing import Any


def _log(level: str, message: str, **fields: Any) -> None:
    payload = {
        "level": level,
        "message": message,
        "ts": datetime.now(timezone.utc).isoformat(),
        **fields,
    }
    print(json.dumps(payload, default=str))


def log_info(message: str, **fields: Any) -> None:
    _log("info", message, **fields)


def log_error(message: str, **fields: Any) -> None:
    _log("error", message, **fields)


# MCP debug logging — enable with MCP_DEBUG=1 or MCP_DEBUG=true.
# Logs raw tool call details, session lifecycle, and auth envelope metadata.
def log_mcp_debug(message: str, data: dict[str, Any] | None = None) -> None:
    if os.getenv("MCP_DEBUG", "") not in ("1", "true", "TRUE", "True"):
        return
    if data is None:
        print("[mcp-debug]", message)
        return
    print("[mcp-debug]", message, json.dumps(data, ensure_ascii=False))
